pkg/dex/store/k8s_store.go

```diff
@@ -0,0 +1,150 @@
+func newKubernetesStore(logger *slog.Logger) (storage.Storage, error) {
+	cfg := kubernetes.Config{InCluster: true}
+	cfgPath := determineKubeMode()
+	if strings.TrimSpace(cfgPath) != "" {
+		cfg.InCluster = false
+		cfg.KubeConfigFile = cfgPath
+	}
+	dexStorage, err := cfg.Open(logger)
+	if err != nil {
+		return nil, err
+	}
+	return dexStorage, nil
+}
+
+func determineKubeMode() string {
+	cfgPath := os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
+	if len(cfgPath) > 0 {
+		return cfgPath
+	}
+	_, err := rest.InClusterConfig()
+	if err == nil {
+		return ""
+	}
+	home := homedir.HomeDir()
+	return filepath.Join(home, ".kube", "config")
+}
+
+func (k *k8sDex) GetBackend() string {
+	return k.backend
+}
+
+// CreateUpdateConnector - creates or updates a dex connector in dex kubernetes storage backend
+func (k *k8sDex) CreateUpdateConnector(ctx context.Context, k8sClient client.Client, org *greenhouseapisv1alpha1.Organization, configByte []byte) (err error) {
+	var result clientutil.OperationResult
+	var dexConnector = new(dexapi.Connector)
+	namespaceName := org.GetName()
+	dexConnector.SetName(namespaceName)
+	dexConnector.SetNamespace(namespaceName)
+	result, err = clientutil.CreateOrPatch(ctx, k8sClient, dexConnector, func() error {
+		dexConnector.DexConnector.Type = dexConnectorTypeGreenhouse
+		dexConnector.DexConnector.Name = cases.Title(language.English).String(org.Name)
+		dexConnector.DexConnector.ID = org.GetName()
+		dexConnector.DexConnector.Config = configByte
+		return controllerutil.SetControllerReference(org, dexConnector, k8sClient.Scheme())
+	})
+	if err != nil {
+		return
+	}
+	switch result {
+	case clientutil.OperationResultCreated:
+		log.FromContext(ctx).Info("created dex connector", "namespace", dexConnector.Namespace, "name", dexConnector.GetName())
+	case clientutil.OperationResultUpdated:
+		log.FromContext(ctx).Info("updated dex connector", "namespace", dexConnector.Namespace, "name", dexConnector.GetName())
+	}
+	return
+}
+
+// CreateUpdateOauth2Client - creates or updates an oauth2 client in dex kubernetes storage backend
+func (k *k8sDex) CreateUpdateOauth2Client(ctx context.Context, k8sClient client.Client, org *greenhouseapisv1alpha1.Organization) error {
+	var oAuth2Client = new(dexapi.OAuth2Client)
+	namespaceName := org.GetName()
+	oAuth2Client.SetName(encodedOAuth2ClientName(namespaceName))
+	oAuth2Client.SetNamespace(namespaceName)
+
+	result, err := clientutil.CreateOrPatch(ctx, k8sClient, oAuth2Client, func() error {
+		oAuth2Client.Client.Public = true
+		oAuth2Client.Client.ID = org.Name
+		oAuth2Client.Client.Name = org.Name
+		for _, requiredRedirectURL := range []string{
+			"http://localhost:8085",
+			"http://localhost:8000",
+			"https://dashboard." + common.DNSDomain,
+			fmt.Sprintf("https://%s.dashboard.%s", org.Name, common.DNSDomain),
+		} {
+			oAuth2Client.Client.RedirectURIs = util.AppendStringToSliceIfNotContains(requiredRedirectURL, oAuth2Client.RedirectURIs)
+		}
```

Comment:
I think I remember why we have this in place. This hack was/is used by the UI Team to add additional redirect URIs to the client, to be used when testing the UI against the IDProxy. As noted on the Postgres implementation, this is not easily done there.
With the current implementation it will require manual modification of the entry in the database.

Specifying additional redirectURIs will be required for the `oauth2-proxy` as well, when using it to secure the service proxy. Let's keep the code in and tackle this in an additional issue #893